app/models/neighbour_response.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+class NeighbourResponse
+  include ActiveModel::API
+  include ActiveModel::Attributes
+  include ActiveModel::Model
+
+  attribute :neighbour_response
+  attribute :name
+  attribute :email
+  attribute :address
+  attribute :response
+  attribute :summary_tag
+  attribute :tags, array: true, default: []
+  attribute :design
+  attribute :use
+  attribute :privacy
+  attribute :light
+  attribute :access
+  attribute :noise
+  attribute :traffic
+  attribute :other
+  attribute :stage
+  attribute :move_next
+  attribute :move_back
+  attribute :planning_application_id, :integer
+
+  STAGES = %w[about_you thoughts response check].freeze
+
+  validates :name, presence: true, if: :about_you_stage?
+  validates :summary_tag, presence: true, if: :thoughts_stage?
+  validate :at_least_one_comment, if: -> { response_stage? && information_not_filled? }
+
+  attr_reader :params
+
+  def initialize(params)
+    super(params)
+    @params = params
+  end
+
+  def stage
+    @stage ||= stage_param.in?(STAGES) ? stage_param : STAGES.first
+  end
+
+  def name
+    response_params[:name]
+  end
+
+  def summary_tag
+    response_params[:summary_tag]
+  end
+
+  def email
+    response_params[:email]
+  end
+
+  def address
+    response_params[:address]
+  end
+
+  def design
+    response_params[:design]
+  end
+
+  def use
+    response_params[:use]
+  end
+
+  def light
+    response_params[:light]
+  end
+
+  def privacy
+    response_params[:privacy]
+  end
+
+  def access
+    response_params[:access]
+  end
+
+  def noise
+    response_params[:noise]
+  end
+
+  def traffic
+    response_params[:traffic]
+  end
+
+  def other
+    response_params[:other]
+  end
+
+  def tags
+    response_params[:tags]
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def save
+    @stage = previous_stage and return false if moving_backwards?
+
+    @stage = stage and return false unless valid?
+
+    @stage = next_stage and return false unless done?
```

Comment:
Is this a clearer expression of the intended logic?

``` ruby
if moving_backwards?
  @stage = previous_stage
elsif invalid?
  @stage = stage
else
  @stage = next_stage
end

return false unless done?
```

Depends on `done?` i guess? Might need another stage added which means on the check page and the form is valid.